hot-desk/src/main/java/com/exadel/sandbox/booking/repository/BookingRepository.java

```diff
@@ -2,6 +2,55 @@
 
 import com.exadel.sandbox.booking.entity.Booking;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.time.LocalDate;
+import java.util.Date;
+import java.util.List;
 
 public interface BookingRepository extends JpaRepository<Booking, Long> {
+    List<Booking> findBookingsBySeatId(Long id);
+    List<Booking> findBookingsByEmployeeId(Long id);
+    List<Booking> findBookingsByParkingSpotId(Long id);
+
+//    Returns 1 if seat free on those dates else 0
+    @Query(nativeQuery = true,
+    value = "select count(distinct s.seat_id)\n" +
+            "from booking b\n" +
+            "         right join seat s on b.seat_id = s.seat_id\n" +
+            "where s.seat_id =:seatId and s.seat_id not in (select b.seat_id\n" +
+            "                                from booking b\n" +
+            "                                         inner join booking_dates bd on b.bo_id = bd.bo_id\n" +
+            "                                where bd.date in (:dates))")
+    Integer checkIfSeatIsFree(Long seatId, List<LocalDate> dates);
```

Comment:
I think it better to use EXIST extend of COUNT, please check other places

Reply:
did you mean this?

select exists(
select *
     from booking b
                     right join employee e on e.em_id = b.em_id
            where e.em_id = :employeeId
              and e.em_id not in (select b.em_id
                                  from booking b
                                           inner join booking_dates bd on b.bo_id = bd.bo_id
                                  where bd.date in (:dates)))

Reply:
yes